src/formatting.ts

```diff
@@ -221,24 +221,96 @@ function toggleListSingleLine(doc: TextDocument, line: number, wsEdit: Workspace
     const lineText = doc.lineAt(line).text;
     const indentation = lineText.trim().length === 0 ? lineText.length : lineText.indexOf(lineText.trim());
     const lineTextContent = lineText.substr(indentation);
+    const currentMarker = getCurrentListStart(lineTextContent);
+    const nextMarker = getNextListStart(currentMarker);
 
-    if (lineTextContent.startsWith("- ")) {
-        wsEdit.replace(doc.uri, new Range(line, indentation, line, indentation + 2), "* ");
-    } else if (lineTextContent.startsWith("* ")) {
-        wsEdit.replace(doc.uri, new Range(line, indentation, line, indentation + 2), "+ ");
-    } else if (lineTextContent.startsWith("+ ")) {
-        wsEdit.replace(doc.uri, new Range(line, indentation, line, indentation + 2), "1. ");
-    } else if (/^\d+\. /.test(lineTextContent)) {
+    // 1. delete current list marker
+    wsEdit.delete(doc.uri, new Range(line, indentation, line, getMarkerEndCharacter(currentMarker, lineText)));
+
+    // 2. insert next list marker
+    if (nextMarker !== empty)
+        wsEdit.insert(doc.uri, new Position(line, indentation), nextMarker);
+}
+
+const empty = "";
+const dash = "- ";
+const star = "* ";
+const plus = "+ ";
+const num = "1. ";
+const numClosingParetheses = "1) ";
```

Comment:
would be better to put them in a dict, e.g. `List_Marker`, otherwise it is confusing to have these names as top-level constants. Also, you may use this `Title_Style` for the consts below.

Reply:
Put those const into a enum.